src/udev/meson.build

```diff
@@ -113,15 +113,30 @@ if get_option('link-udev-shared')
         udev_rpath = rootlibexecdir
 else
         udev_link_with = [libshared_static,
-                          libsystemd_internal]
+                          libsystemd_static]
         udev_rpath = ''
 endif
 
-libudev_internal = static_library(
+libudev_static = static_library(
         'udev',
         libudev_sources,
         include_directories : includes,
-        link_with : udev_link_with)
+        link_with : udev_link_with,
+        c_args : ['-fvisibility=default'])
+
+libudev = shared_library(
+        'udev',
+        'udev.h', # pick a header file at random to work around old meson bug
+        version : libudev_version,
+        include_directories : includes,
+        link_args : ['-shared',
+                     '-Wl,--version-script=' + libudev_sym_path],
+        link_with : udev_link_with,
```

Comment:
Is this correct?

Reply:
I guess this should be `libsystemd`. Right?

Reply:
Removing this is not sufficient to unbreak libudev when `link-udev-shared=true`. The problem is that pulling libudev_static pulls in libsystemd-shared which is wrong for libudev. Something like this is needed:

```diff
diff --git a/src/udev/meson.build b/src/udev/meson.build
index 6516307db..e2e1aa0fe 100644
--- a/src/udev/meson.build
+++ b/src/udev/meson.build
@@ -117,11 +117,18 @@ else
         udev_rpath = ''
 endif

+libudev_basic = static_library(
+        'udev-basic',
+        libudev_sources,
+        include_directories : includes,
+        c_args : ['-fvisibility=default'])
+
 libudev_static = static_library(
         'udev',
-        libudev_sources,
+        'udev.h',
         include_directories : includes,
         link_with : udev_link_with,
+        link_whole : libudev_basic,
         c_args : ['-fvisibility=default'])

 libudev = shared_library(
@@ -131,7 +138,8 @@ libudev = shared_library(
         include_directories : includes,
         link_args : ['-shared',
                      '-Wl,--version-script=' + libudev_sym_path],
-        link_with : udev_link_with,
+        link_with : [libsystemd_static, libshared_static],
         link_whole : libudev_static,
         dependencies : [threads],
         link_depends : libudev_sym,
```

```
% ldd src/udev/libudev.so.1.6.8
        linux-vdso.so.1 (0x00007ffece712000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f46ca094000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f46c9cf1000)
        /lib64/ld-linux-x86-64.so.2 (0x000055ac87e02000)
```

Reply:
Indeed, my guess was wrong.